src/mds/objects.py

```diff
@@ -384,6 +385,51 @@ async def get_object(guid: str, request: Request) -> JSONResponse:
     return JSONResponse(response, HTTP_200_OK)
 
 
+@mod.delete("/objects/{guid:path}")
+async def delete_object(
+    guid: str, request: Request, token: HTTPAuthorizationCredentials = Security(bearer)
+) -> JSONResponse:
+    """
+    Delete the metadata for the specified object. Remove the object from
+    existing bucket location(s) by proxying to fence DELETE /data/file_id.
+    Uses the response status code from fence to determine whether user has
+    permission to delete metadata.
+
+    Args:
+        guid (str): indexd GUID or alias, or MDS key
+        request (Request): starlette request (which contains reference to FastAPI app)
+    Returns:
+        204: if record and metadata are deleted
+        403: if fence returns a 403 unauthorized response
+        500: if fence does not return 204 or 403 or there is an error deleting metadata
+    """
+    try:
+        fence_endpoint = (
+            config.DATA_ACCESS_SERVICE_ENDPOINT.rstrip("/") + "/data/" + guid
```

Comment:
Maybe a good spot for `urljoin` https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin

Reply:
Agreed! Looks much cleaner with this change